datacube/utils/math.py

```diff
@@ -119,6 +119,14 @@ def num2numpy(x, dtype, ignore_range=None):
     return None
 
 
+def data_resolution(data):
+    if data.size == 1:
```

Comment:
`data.size < 2` is better

Reply:
It was hoped that when `data.size`=0, it would go on the execution and fail at `data[-1]`.

Reply:
why should 0 behave differently than 1, both don't have enough information to compute resolution, both should fail in equivalent ways, raising IndexError in this situation is not a good form

Reply:
'cause if `data.size=1`, can still try to locate the information from attributes to continue. If `data.size=0`, it doesn't make sense to continue and supposed to fail hard. They should be dealt differently. Letting it raising an IndexError is me being lazy... -_-

Reply:
that's why `<2` is a better check: not enough data to compute resolution as far as this function goes, what you can and can not do with 1 or 0 sized arrays outside of this function has no influence on the implementation of this function.

Reply:
With the argument here, maybe the check should have done before calling this function then. Fail sooner better, I guess.

Reply:
it's ok to check things twice, this is not some performance crucial inner loop function,

also maybe this function shouldn't exist, or should be some inner hidden function elsewhere.